### PR TITLE
Allowing usernames, passwords and database names with special characters (percent encoded in URI)

### PR DESCRIPTION
**Pull request by MartinManev**

### CODE REVIEW

File: lib/src/postgresql_impl/connection.dart [hunk 1]
```diff
@@ -10,7 +10,7 @@ class ConnectionImpl implements Connection {
       TypeConverter typeConverter,
       String getDebugName())
     : _userName = settings.user,
-      _passwordHash = _md5s(settings.password + settings.user),
+      _passwordHash = _md5s(Uri.decodeFull(settings.password) + settings.user),
```
Comment (Fox32) [inline lib/src/postgresql_impl/connection.dart]: Would this be better placed in the settings class? What about decoding the user name?

Comment (MartinManev) [inline lib/src/postgresql_impl/connection.dart]: Thanks for the input. I've moved the encoding/decoding to the settings class as you suggested.

**Comment by Fox32** [comment]
Looks good

**Comment by xxgreg** [comment]
Hi Martin,

Thanks for the PR, and sorry that it's taken me so long to get to it.

I think I botched the fix for #67, which is related to this. I've taken your code and changed it a bit and put it on a branch here:

https://github.com/xxgreg/dart_postgresql/compare/ft-settings-uri-encoding

Does this solve your problem?

If you or Oliver have time to try this out, it would be much appreciated.

Cheers

**Comment by pisabev** [comment]
Hi Greg,

Martin and I work in a team. The branch is perfect... 

Cheers

**Comment by xxgreg** [comment]
Cool. I've published 0.3.3 to pub.

**Comment by MartinManev** [comment]
Unfortunately, the current implementation with the Uri class does not encode all special characters: ':/@?#[] and some others are left untouched and cause an exception, while postgres does accept them as long as they are properly encoded. This is exactly the reason why I used Uri.encodeComponent() on each individual property, as this function encodes all of the special characters.

**Pull request closed without merging**